### PR TITLE
Feature/202 - Alibaba ram user password lowercase policy plugin with test cases

### PR DESCRIPTION
**Pull request by umermehmood**

### CODE REVIEW

**Review by dipsubha06 — APPROVED** [review]

File: collectors/alibaba/collector.js [hunk 1]
```diff
@@ -31,6 +31,8 @@ var regionEndpointMap = {
         'ap-northeast-1', 'ap-south-1', 'eu-central-1', 'eu-west-1', 'me-east-1']
 };
 
+var apiVersion = '2015-05-01';
```
Comment (giorod3) [inline collectors/alibaba/collector.js:34]: we're removing this from here and adding it to each call object

File: collectors/alibaba/collector.js [hunk 2]
```diff
@@ -262,7 +266,7 @@ var collect = function(AlibabaConfig, settings, callback) {
                     let endpoint = (regionEndpointMap[serviceLower] && regionEndpointMap[serviceLower].includes(region)) ?
                         `https://${serviceLower}.${region}.aliyuncs.com` : `https://${serviceLower}.aliyuncs.com`;
                     LocalAlibabaConfig['endpoint'] = endpoint;
-                    LocalAlibabaConfig['apiVersion'] = callObj.apiVersion;
+                    LocalAlibabaConfig['apiVersion'] = callObj.apiVersion || apiVersion ;
```
Comment (giorod3) [inline collectors/alibaba/collector.js:269]: same here

File: collectors/alibaba/collector.js [hunk 3]
```diff
@@ -373,7 +377,7 @@ var collect = function(AlibabaConfig, settings, callback) {
                         } else {
                             LocalAlibabaConfig['endpoint'] = (regionEndpointMap[serviceLower] && regionEndpointMap[serviceLower].includes(region)) ?
                                 `https://${serviceLower}.${region}.aliyuncs.com` : `https://${serviceLower}.aliyuncs.com`;
-                            LocalAlibabaConfig['apiVersion'] = callObj.apiVersion;
+                            LocalAlibabaConfig['apiVersion'] = callObj.apiVersion || apiVersion;
```
Comment (giorod3) [inline collectors/alibaba/collector.js:380]: same

**Review by giorod3 — APPROVED** [review]
@umermehmood looks good! can you remove the apiVersion logic and this is good to merge

File: collectors/alibaba/collector.js (hunk 1 above)
Comment (umermehmood) [reply collectors/alibaba/collector.js:34]: Sure, will update it accordingly.

File: collectors/alibaba/collector.js [hunk 4]
```diff
@@ -188,6 +189,8 @@ var postcalls = [
                 reliesOnCall: 'ListUsers',
                 filterKey: ['UserName'],
                 filterValue: ['UserName'],
+                resultFilter: 'User',
+                apiVersion: '2015-05-01',
```
Comment (AkhtarAmir) [inline collectors/alibaba/collector.js:193]:
```suggestion
                apiVersion: '2015-05-01',
```

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

**Review by giorod3 — APPROVED** [review]

**Pull request merged**